qutip/operators.py

```diff
@@ -422,17 +423,65 @@ def create(N, offset=0):
     return qo.dag()
 
 
+def _implicit_tensor_dimensions(dimensions):
```

Comment:
Dimensions are not just a list of integer but can be nested list for superoperator:
```
>>> liouvillian(qeye(2)).dims
[[[2], [2]], [[2], [2]]]
```
There are function in qutip.dimensions that could be useful like `flatten`

Reply:
Ah, thanks for pointing me to `qutip.dimensions` - I'd not come across that yet.  I'm still working on getting more familiar with the internals.